File: 404-sum-of-left-leaves/404.py
class Solution:
    def dfs(self, root, isleft):
        if root is None:
            return
        
        self.dfs(root.left, True)
        self.dfs(root.right, False)
        
        if root.left is None and root.right is None and isleft:
            self.sum += root.val
        
        return 
        
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        self.sum = 0
        self.dfs(root, False)
        return self.sum

File: 404-sum-of-left-leaves/404.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    int dfs(TreeNode* node, bool isLeft) {
        if (!node) {
            return 0;
        }
        int left = dfs(node->left, true);
        int right = dfs(node->right, false);
        if (!node->left && !node->right && isLeft) {
            return node->val + left + right;
        } else {
            return left + right;
        }
    }
public:
    int sumOfLeftLeaves(TreeNode* root) {
        return dfs(root, false);
    }
};
